Framework/PythonInterface/plugins/algorithms/HB2AReduce.py

```diff
@@ -0,0 +1,274 @@
+            if os.path.isfile(exclude_filename):
+                with warnings.catch_warnings():
+                    warnings.simplefilter("ignore")
+                    exclude_detectors = np.loadtxt(exclude_filename, ndmin=1, dtype=int)
+            else:
+                exclude_detectors=np.empty(0, dtype=int)
+        else:
+            exclude_detectors = np.array(self.getProperty("ExcludeDetectors").value)
+        if len(exclude_detectors) > 0:
+            logger.notice("Excluding anodes: {}".format(exclude_detectors))
+        detector_mask[exclude_detectors-1] = False
+        return detector_mask
+
+    def get_vanadium(self, detector_mask, m1, colltrans, exp, indir):
+        """
+        This function returns either (vanadium_count, vanadium_monitor, None) or
+        (None, None, vcorr) depending what type of file is provided by getProperty("Vanadium")
+        """
+        if not self.getProperty("Normalise").value:
+            return None, None, np.ones(44)[detector_mask]
+
+        vanadium_filename = self.getProperty("Vanadium").value
+        if vanadium_filename:
+            if vanadium_filename.split('.')[-1] == 'dat':
+                vanadium = np.genfromtxt(vanadium_filename)
+                vanadium_count = vanadium[:, 5:49].sum(axis=0)[detector_mask]
+                vanadium_monitor = vanadium[:, 3].sum()
+                logger.notice("Using vanadium data file: {}".format(vanadium_filename))
+                return vanadium_count, vanadium_monitor, None
+            else:
+                vcorr_filename = vanadium_filename
+        else: # Find adjacent vcorr file
+            # m1 = 0 -> Ge 115, 1.54A
+            # m1 = 9.45 -> Ge 113, 2.41A
+            # colltrans = 0 -> IN
+            # colltrans = +/-80 -> OUT
+            vcorr_filename = 'HB2A_{}__Ge_{}_{}_vcorr.txt'.format(exp,
+                                                                  115 if np.isclose(m1, 0, atol=0.1) else 113,
+                                                                  "IN" if np.isclose(colltrans, 0, atol=0.1) else "OUT")
+        vcorr_filename = os.path.join(indir, vcorr_filename)
+        logger.notice("Using vcorr file: {}".format(vcorr_filename))
+        if not os.path.isfile(vcorr_filename):
+            raise RuntimeError("Vanadium file {} does not exist".format(vcorr_filename))
+
+        return None, None, np.genfromtxt(vcorr_filename)[detector_mask]
+
+    def process(self, counts, scale, monitor, vanadium_count=None, vanadium_monitor=None, vcorr=None):
+        """Reduce data not binning"""
+        if vcorr is not None:
+            y = counts/vcorr[:, np.newaxis]/monitor
+            e = np.sqrt(counts)/vcorr[:, np.newaxis]/monitor
+        else:
+            y = counts/vanadium_count[:, np.newaxis]*vanadium_monitor/monitor
+            e = np.sqrt(1/counts + 1/vanadium_count[:, np.newaxis] + 1/vanadium_monitor + 1/monitor)*y
+        return y*scale, e*scale
```

Comment:
Should replace `nan` in individual spectra with zeros to make plots make more sense when `monitor=0`.